src/math/crypto/curve/ec_point.zig

```diff
@@ -236,4 +259,158 @@ pub const AffinePoint = struct {
             .infinity = false,
         };
     }
+
+    /// Doubles a point on an elliptic curve with the equation y^2 = x^3 + alpha*x + beta.
+    ///
+    /// # Arguments
+    /// - `self` - The point.
+    /// - `alpha` - The alpha parameter of the elliptic curve.
+    ///
+    /// # Returns
+    /// The doubled elliptic curve point.
+    pub fn ecDouble(self: *Self, alpha: Felt252) ECError!AffinePoint {
```

Comment:
Here I would say that it is better to mark `alpha` as comptime as long as we know it as a constant of each curve. Or maybe we can put it as an additional field of the struct.

Reply:
i decided to make it part of the struct because there were errors with having it be comptime for example:
        doubled_point = try doubled_point.ecDouble(alpha);

my issue now is with adding all of the .alpha to the gen/constants.zig